http/src/main/java/io/micronaut/http/converters/SharedHttpConvertersRegistrar.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.converters;
+
+import io.micronaut.context.exceptions.ConfigurationException;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.convert.MutableConversionService;
+import io.micronaut.core.convert.TypeConverterRegistrar;
+import io.micronaut.http.HttpStatus;
+import io.micronaut.http.HttpVersion;
+
+import java.net.InetSocketAddress;
+import java.net.MalformedURLException;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URL;
+import java.util.Optional;
+
+/**
+ * Converter registrar for HTTP classes.
+ *
+ * @author Denis Stepanov
+ * @since 4.0.0
+ */
+@Internal
+public class SharedHttpConvertersRegistrar implements TypeConverterRegistrar {
```

Comment:
make `final`. Does this need to be public?

Reply:
I think it needs to be public for the service loader.